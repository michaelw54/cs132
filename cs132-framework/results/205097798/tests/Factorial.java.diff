1,4c1
< ParseException: Encountered " <IDENTIFIER> "w_2 "" at line 2, column 1.
< Was expecting:
<     "return" ...
<     
---
> 3628800
