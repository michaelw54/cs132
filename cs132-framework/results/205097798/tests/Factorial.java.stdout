func Main()
    w4 = 4
    w2 = alloc(w4)
    if0 w2 goto null_2
    goto passNull_2
  null_2:
    error("null pointer")
  passNull_2:
    w4 = 4
    w3 = alloc(w4)
    if0 w3 goto null_3
    goto passNull_3
  null_3:
    error("null pointer")
  passNull_3:
    [w2 + 0] = w3
    w4 = @Fac_ComputeFac
    [w3 + 0] = w4
    w1 = [w2 + 0]
    w1 = [w1 + 0]
    w5 = 10
    w0 = call w1(w2 w5)
    print(w0)
    ret_ = 0
    return ret_
    
func Fac_ComputeFac(this num)
    num_aux = 0
    w8 = num
    w10 = 1
    w7 = w8 < w10
    if0 w7 goto else_7
    w12 = 1
    num_aux = w12
    goto end_7
  else_7:
    w15 = num
    w19 = this
    if0 w19 goto null_19
    goto passNull_19
  null_19:
    error("null pointer")
  passNull_19:
    w18 = [w19 + 0]
    w18 = [w18 + 0]
    w21 = num
    w23 = 1
    w20 = w21 - w23
    w17 = call w18(w19 w20)
    w14 = w15 * w17
    num_aux = w14
  end_7:
    w28 = num_aux
    return w28
    
